core/src/rust_types.rs

```diff
@@ -314,7 +317,7 @@ impl Display for SpecialRustType {
 #[derive(Debug, Error)]
 #[allow(missing_docs)]
 pub enum RustTypeParseError {
-    #[error("{0:?}")]
+    #[error("Unsupported type: \"{}\"", .0.iter().join(","))]
```

Comment:
Small organization question: Would it be confusing to move this into `error.rs` along side the other parse errors? It seems the `with_span` method is defined there, so it feels like it might be worthwhile to have this there as well for consistency.